### ISSUE TITLE
Add option to configure log level for missing attributes

### ISSUE THREAD

**Issue by rocec** [asciidoctor/asciidoctor#4155]
The value of the attribute `attribute-missing` determines how Asciidoctor treats references to undefined attributes. Currently, four options are supported: either take one of three silent actions (skip, drop, drop-line), or issue a warning and skip it.

For a strict quality control, treating warnings as errors is often desirable. It would therefore be useful to make the log level of these warnings configurable, and thus be able to issue errors instead of warnings.

As @mojavelinux suggested in a [discussion](https://asciidoctor.zulipchat.com/#narrow/stream/279642-users/topic/Warnings.20as.20errors.20in.20the.20IntelliJ.20plugin/near/251514579) on Zulip:

> I'd also be willing to change core so that attribute-missing is "log" instead of "warn" and another attribute could set the level...like attribute-missing-log-level=error (or something like that).

**Comment by mojavelinux** [comment]
I agree this is the right progression. When the attribute-missing attribute was first introduced, there was no logging facility in Asciidoctor. Therefore, the only way to to report the message was to output a warning to stderr (hence the value, "warn"). Now that Asciidoctor has a logging facility, and this message gets directed to that log, it makes more sense to identify the action as "log" and then somehow allow the level to be configured.

**Comment by oddhack** [comment]
We would also find this change useful. We'd like to treat missing attribute as errors, but cannot turn on --failure-level=WARNING globally as there are other types of warnings in our builds which are acceptable, for now.
